bindings/gumjs/gumv8core.cpp

```diff
@@ -2276,52 +2284,85 @@ gum_v8_native_function_invoke (GumV8NativeFunction * self,
 {
   auto core = (GumV8Core *) info.Data ().As<External> ()->Value ();
   auto isolate = core->isolate;
-  gsize num_args_required = self->cif.nargs;
+  auto cif = &self->cif;
+  gsize num_args_declared = cif->nargs;
   gsize num_args_provided = (argv != nullptr) ? argc : info.Length ();
+  gsize num_args_fixed = self->nargs_fixed;
+  gboolean is_variadic = self->is_variadic;
 
-  if (num_args_provided != num_args_required)
+  if ((is_variadic && num_args_provided < num_args_fixed) || (!is_variadic && num_args_provided != num_args_declared))
   {
     _gum_v8_throw_ascii_literal (isolate, "bad argument count");
     return;
   }
 
-  auto rtype = self->cif.rtype;
-
+  auto rtype = cif->rtype;
+  auto atypes = cif->arg_types;
   gsize rsize = MAX (rtype->size, sizeof (gsize));
   gsize ralign = MAX (rtype->alignment, sizeof (gsize));
   auto rvalue = (GumFFIValue *) g_alloca (rsize + ralign - 1);
   rvalue = GUM_ALIGN_POINTER (GumFFIValue *, rvalue, ralign);
 
   void ** avalue;
   guint8 * avalues;
+  ffi_cif tmp_cif;
+  GumFFIValue tmp_value = { 0, };
 
-  if (num_args_required > 0)
+  if (num_args_provided > 0)
   {
-    avalue = (void **) g_alloca (num_args_required * sizeof (void *));
+    avalue = (void **) g_alloca (MAX (num_args_declared, num_args_provided) * sizeof (void *));
```

Comment:
Exceeds 80 cols. (Also spotted this elsewhere: please review this across the PR.)

Reply:
ok, i will fix it.